go/pkg/client/cas.go

```diff
@@ -208,50 +202,53 @@ func getUnifiedLabel(labels map[string]bool) string {
 }
 
 // capToLimit ensures total length does not exceed max header size.
-func capToLimit(m *requestMetadata, limit int) {
-	total := len(m.toolName) + len(m.actionID) + len(m.invocationID)
+func capToLimit(m *ContextMetadata, limit int) {
```

Comment:
Don't you have to now account for all of ContextMetadata here?

Reply:
That would be difficult, so I still only do the merging and capping for the two fields, ActionID and InvocationID specifically. I even regress combining the tool name here (because it made little sense anyway).

Reply:
I agree it's difficult (maybe hacky?), but like `m.ToolName`, we have to account for it when checking the full header size?

Reply:
Right, sorry! Done.